src/utils/include/utils/Histogram.hpp

```diff
@@ -153,19 +159,19 @@ class Histogram {
    * \brief Check if the position lies within the histogram limits.
    * \param pos     Position to check.
    */
-  bool check_limits(std::span<const U> pos) const {
+  bool check_limits(std::span<const U> const &pos) const {
     assert(pos.size() == M);
-    bool within_range = true;
-    for (std::size_t i = 0; i < M; ++i) {
-      if (pos[i] < m_limits[i].first or pos[i] >= m_limits[i].second)
-        within_range = false;
-    }
-    return within_range;
+    auto it_limits = m_limits.begin();
+    return std::ranges::all_of(pos, [&it_limits](U const value) {
+      auto const [lower, upper] = *it_limits;
+      ++it_limits;
+      return value >= lower and value <= upper;
```

Comment:
Looking at the code, previously, we used $[\mathrm{lower}, \mathrm{upper})$ as the valid range. No idea if there are any side effects if we leave it in...

```suggestion
      return value >= lower and value < upper;
```

Reply:
good point, this boundary check was fixed and is now properly documented in the user guide